sstables/mx/writer.cc

```diff
@@ -1003,6 +1004,7 @@ void writer::consume(tombstone t) {
 
     if (t) {
         _collector.update_min_max_components(clustering_key_prefix::make_empty(_schema));
+        _has_partition_tombstones = has_partition_tombstones::Yes;
```

Comment:
Wait. We wanted this new metadata to be sure to include an sstable that has a tombstone early. But the previous line (update_min_max_components) will already do that! Filtering based on clustering key will include it early, since the clustering range for this sstable will be empty.

Reply:
On the contrary, the clustering range will be the full range (`before_all_clustered`, `after_all_clustered`). This is what empty clustering key prefix denotes.

If we would set the sstable's clustering range to be empty on a partition tombstone, the current sstable reading functions would give incorrect results because they would not include sstables which have partition tombstones (the ck filter would filter them out).

If the range contains at least one mutation fragment in the partition, the range must include this fragment's position. Since partition tombstones are "everywhere" - they affect all other positions - if the sstable contains a partition tombstone, its range must be the full range.

Reply:
> We wanted this new metadata to be sure to include an sstable that has a tombstone early

This metadata is used to determine whether or not we should use the new TWCS-specific sstable-reader-merger. The merger only works correctly when there are no partition tombstones in the merged sstables. If there may be (which we check using this metadata), we use the old merger.

Reply:
I guess we could use the following method:
- use the new merger only if the clustering range of the sstable is not full.

Reason: it is safe to use the new merger if the clustering range is not full, because that means that it doesn't have partition tombstones.

However I'm not sure if this works for all sstables -- created by Cassandra or an older version of Scylla, i.e. if they enforce this condition -- so the safer condition would be:
- use the new merger only if the clustering range of the sstable is not full, the version is `>= md`, and the sstable has a Scylla component.